airflow_tecton/operators/tecton_job_operator.py

```diff
@@ -46,6 +49,7 @@ def __init__(
         start_time: Union[str, datetime.datetime] = "{{ data_interval_start }}",
         end_time: Union[str, datetime.datetime] = "{{ data_interval_end }}",
         allow_overwrite: bool = False,
+        df_generator: Callable[[], pandas.DataFrame] = None,
```

Comment:
start_time/end_time and df_generator are mutually exclusive. so they should be validated against

Reply:
also we need to allow this function to take in context and args/kwargs. I would look at how bashoperator or pythonoperator do it

Reply:
PythonOperator just uses Callable, so I followed it.

Reply:
added 
```
        op_args: Union[Collection[Any], None] = None,
        op_kwargs: Union[Mapping[str, Any], None] = None,
        templates_dict: Union[dict, None] = None,
        templates_exts: Union[list[str], None] = None,
```
as pythonoperator does